### DISCUSSION TITLE
ESP32 with ILI9488 White screen

### DISCUSSION THREAD

**Discussion by Enigma644** [Bodmer/TFT_eSPI#2309] [Q&A - General] [answered]
I'm having trouble getting anything to display and I could really use some help!

None of the examples work. All I get is the LCD backlight turns on. No graphics.

* ESP32 Dev - https://www.amazon.ca/gp/product/B07NPJJ8ZC
* ILI9488 TFT - https://www.amazon.ca/dp/B08CN6M9VC

User_Setup.h
```C
#define ILI9488_DRIVER     // WARNING: Do not connect ILI9488 display SDO to MISO if other devices share the SPI bus (TFT SDO does NOT tristate when CS is high)

//ESP32
#define TFT_BL   32            // LED back-light control pin
#define TFT_BACKLIGHT_ON HIGH  // Level to turn ON back-light (HIGH or LOW)

#define TFT_MISO 19  //MISO
#define TFT_MOSI 23  //MOSI
#define TFT_SCLK 18  //SCK
#define TFT_CS   15  // Chip select control pin
#define TFT_DC    2  // Data Command control pin
#define TFT_RST   4  // Reset pin (could connect to RST pin)
#define TOUCH_CS 21     // Chip select pin (T_CS) of touch screen

#define LOAD_GLCD   // Font 1. Original Adafruit 8 pixel font needs ~1820 bytes in FLASH
#define LOAD_FONT2  // Font 2. Small 16 pixel high font, needs ~3534 bytes in FLASH, 96 characters
#define LOAD_FONT4  // Font 4. Medium 26 pixel high font, needs ~5848 bytes in FLASH, 96 characters
#define LOAD_FONT6  // Font 6. Large 48 pixel font, needs ~2666 bytes in FLASH, only characters 1234567890:-.apm
#define LOAD_FONT7  // Font 7. 7 segment 48 pixel font, needs ~2438 bytes in FLASH, only characters 1234567890:-.
#define LOAD_FONT8  // Font 8. Large 75 pixel font needs ~3256 bytes in FLASH, only characters 1234567890:-.

#define LOAD_GFXFF  // FreeFonts. Include access to the 48 Adafruit_GFX free fonts FF1 to FF48 and custom fonts
#define SMOOTH_FONT

#define SPI_FREQUENCY  27000000
#define SPI_READ_FREQUENCY  16000000
#define SPI_TOUCH_FREQUENCY  2500000
```

I'm sure I've got the wiring correct, but here's some pics just in-case:
![image](https://user-images.githubusercontent.com/29318/211230076-eb1e06de-30a9-4ce4-aab5-6b1fbdc75a49.png)

![PXL_20230109_014125805](https://user-images.githubusercontent.com/29318/211229990-b1179355-e75d-45fa-818f-0e992bc2aab5.jpg)
![PXL_20230109_014116900](https://user-images.githubusercontent.com/29318/211229996-815fc449-8057-4928-910f-0ba549bc0495.jpg)
![PXL_20230109_014220012](https://user-images.githubusercontent.com/29318/211230001-8124806b-38d0-43e5-b460-2b5e0f863578.jpg)
![PXL_20230109_014231261](https://user-images.githubusercontent.com/29318/211230008-79e2f62b-9c27-4544-8d40-8c44c143d00f.jpg)

**Comment by Bodmer** [comment]
Run the Read_User_Setup example and look at the serial monitor output. Check the correct setup information is being picked up by the compiler.  That display + processor combination has been tested and should work.

The soldering on the header of the display looks a bit dodgey so it would be correcting that.

**Reply by EngenheiroGamer** [reply]
I've just had a problem with this same display and my ESP32 WROOM 32D (DevKitCv4), and your advice worked fine for me. Probably was getting poor contact through pcb and wiring, but using this example to be sure of the right pinout did its job!

**Comment by Enigma644** [comment]
Read_User_Setup output is correct. I applied extra solder to all the pin heads and tested continuity from the pin base to ICs/pads and all are working.
I would also note that this is the second screen I've got from the Amazon reseller after I asked for a replacement. They both have dodgy looking pin headers, but I think they're all connected. As I have two, I carefully cut the adhesive off the screen on this one to check the pin continuity for the pins that only go to the screen and have no test points on the back... all are fine. Screen still only shows LED backlight though :-/

I would also add that the touch screen is working so there's that.

Is it possible this isn't ILI9488? I would have thought if it was mis-labelled, I'd at least get some junk pixel flashing going on.

![PXL_20230109_230727524](https://user-images.githubusercontent.com/29318/211426305-6070f716-5e4f-4d44-98b5-270e8ddd27a0.jpg)
![PXL_20230109_230715801](https://user-images.githubusercontent.com/29318/211426311-52da459e-4a83-463a-ae74-849943a77b33.jpg)

**Comment by Bodmer** [comment]
I can't see anything wrong.

I suggest you try different drivers to see if it is mislabelled.  ILI9486_DRIVER and ILI9481_DRIVER would be good to try.

**Comment by Enigma644** [comment]
Nothing from either of those. :-/
Want me to post it to you for your collection? If so dm me your address. If you can get it working, let me know and it's twin can be set free from the junk parts bin!

**Comment by Bodmer** [comment]
What versions are you using of ESP32 board package, Arduino IDE and TFT_eSPI?

**Reply by Enigma644** [reply]
ESP32: v2.0.6
Arduino IDE v1.8.13 because I prefer the UI. Tested with v2.0.3, no difference.
TFT_eSPI: 2.4.79

**Comment by Bodmer** [comment]
Interestingly[ in this post ](https://github.com/Bodmer/TFT_eSPI/discussions/1892)a user has bought the same display part number from amazon and it did not behave like a normal ILI9488 screen. This suggests it is a different controller or a Chinese clone chip perhaps.

Maybe try a lower SPI clock rate? Typically an ILI9488 will run up to 80MHz but if it is a different chip it may have a lower maximum clock rate.

**Reply by Enigma644** [reply]
I've tried SPI_FREQUENCY 27000000, 20000000, 15000000, 10000000, 5000000. No change.

**Reply by Bodmer** [reply]
Last hope: The display has a 3.3V regulator, if it is not a low dropout type the display mya not be getting the right voltage. Try connecting the display VCC to 5V, probably that is VIN on the ESP32 board.

Many users have sucessfully used the ILI9488 with the ESP32, so now I have run out of ideas as to why it does not work.

**Comment by Enigma644** [comment]
No difference :( I've contacted the Amazon seller asking them to link me to a lib that they know is 100% working. I'll post back here when they get back to me and let you know what happens either way.

**Reply by Rongit53** [reply]
Sorry for that question - I'm new at githhub:  How to get help for another issue for ESP32 with ILI9488 using the TFT-eSPI library?

**Comment by Enigma644** [comment, marked as answer]
They got back to me and said they're giving me a refund.. Didn't even bother trying to link to a library! I guess it's a cheap knock-off and they have zero confidence in it? I'm sorry for taking up your time on this as it was a dead end, but I really aprecheate your effort to resolve this!
